src/InputGroup/_InputGroupAddon.js

```diff
@@ -0,0 +1,48 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const InputGroupAddon = ({
+    addonClassNames,
+    children,
+    className,
+    compact,
+    disableStyles,
+    isButton,
+    ...otherProps
+}) => {
+
+    const addonClasses = classnames(
+        addonClassNames,
+        'fd-input-group__addon',
+        { 'fd-input-group__addon--button': isButton },
+        { 'fd-input-group__addon--button--compact': compact && isButton }
+    );
+
+    return (
+        <span className={addonClasses} {...otherProps} >
+            {children}
+        </span>
+    );
+};
+
+InputGroupAddon.displayName = 'InputGroupAddon';
+
+InputGroupAddon.propTypes = {
+    addonClassNames: PropTypes.string,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    compact: PropTypes.bool,
+    customStyles: PropTypes.object,
```

Comment:
since this isn't wrapped with `withStyles`, it shouldn't have `customStyles` or `disableStyles` props as they wouldn't do anything